chore(tooling): Add .graphml diff Python script

Dev2150 opened the pull request:
Introduces a Python script, compare_develoment_roadmaps.py, designed to analyze and report the differences between two versions of .graphml files (currently designed for the develoment roadmap)

The script helps developers quickly identify:
- Added, removed, and common nodes (edges)
- Changes in node attributes
- Structural differences in the graph

When it comes to revisions it can be used:
- without any arguments (none specified)
- with one revision, e.g. HEAD
- with 2 revisions, e.g.
  - HEAD~1 HEAD~3
  - a1b2c3d 9f8e7d6

This tool is essential for tracking the evolution of graphs and for debugging issues that arise from graph modifications and for verifying changes before they are merged.

The tool is used by triggering a command such as 'git diff -- "MEIOUandTaxes/Documentation/Development roadmap.graphml"'

In order to make use of the tool even easier by using alias like:
- diff-roadmap
- diff-roadmap HEAD~3
- diff-roadmap HEAD~1 HEAD~3
- diff-roadmap a1b2c3d 9f8e7d6

the following lines are to be appended to the file .git/config: "[alias]
 	diff-roadmap = "!sh -c 'git diff \"$@\" -- \"MEIOUandTaxes/Documentation/Development roadmap.graphml\"' :"
"

Dev2150 committed:
chore(tooling): Add graph comparison python script

Introduces a Python script, compare_develoment_roadmaps.py, designed to analyze and report the differences between two versions of .graphml files (currently designed for the develoment roadmap)

The script helps developers quickly identify:
- Added, removed, and common nodes (edges)
- Changes in node attributes
- Structural differences in the graph

When it comes to revisions it can be used:
- without any arguments (none specified)
- with one revision, e.g. HEAD
- with 2 revisions, e.g.
  - HEAD~1 HEAD~3
  - a1b2c3d 9f8e7d6

This tool is essential for tracking the evolution of graphs and for debugging issues that arise from graph modifications and for verifying changes before they are merged.

The tool is used by triggering a command such as 'git diff -- "MEIOUandTaxes/Documentation/Development roadmap.graphml"'

In order to make use of the tool even easier by using alias like:
- diff-roadmap
- diff-roadmap HEAD~3
- diff-roadmap HEAD~1 HEAD~3
- diff-roadmap a1b2c3d 9f8e7d6

the following lines are to be appended to the file .git/config:
"[alias]
 	diff-roadmap = "!sh -c 'git diff \"$@\" -- \"MEIOUandTaxes/Documentation/Development roadmap.graphml\"' :"
"

Dev2150 commented:
My current concern is the location of the script **compare_develoment_roadmaps.py**, since it does not describe the plan and design of the project (so it would not fit in /Documentation), but instead it relates to the maintenance of the development process.

MichaHelder reviewed (commented):
Looks good to me but this Python code is above my non-pay grade.
Will ping Jarkami in Discord for a proper review.

BasileusBasileon reviewed (changes requested):
Haven't used Python in a long time.

Please delete this or comment it properly, unused code should never persist without a good reason and if so should be documented:

https://github.com/MEIOU-and-Taxes/MnT-EU5/blob/b28f32d0bd9598f6f962e3e4bf4f33ced2a00225/MEIOUandTaxes/Documentation/compare_development_roadmaps.py#L210-L220 

Otherwise CR seems good to me. Don't have time to run to QA right now.

Dev2150 committed:
refactor: Split .graphml tooling files & moved to tools/ (for .git-blame-ignore-revs)

Dev2150 committed:
test: Make graphml comparer debuggable via environment variable

Dev2150 committed:
feat: Compare parents & show movement. Make comparisons more stable. Split code

Previous revision's comparison was unstable because upon .graphml edit, reindexing happens. The fix is to assign ID custom parameter to all nodes.

Dev2150 committed:
refactor(tooling): Move graphml scripts to separate repository

Dev2150 committed:
docs(roadmap): Add custom properties, e.g. ID, status, finished_date, priority, category, effort

Dev2150 commented:
Moved logic to another repo
